lua/entities/pewpew_base_cannon/init.lua

```diff
@@ -205,11 +209,15 @@ end
 function ENT:NewSystem_FireBullet()
 	local Dir, Pos = pewpew:GetFireDirection( self.Direction, self, ent )
 	local num = self.Bullet.Spread or 0
-	if (num and num != 0) then
-		local randomang = Angle( math.Rand(-num,num), math.Rand(-num,num), math.Rand(-num,num) )
-		Dir:Rotate(randomang)
-	end
-	pewpew:FireBullet( Pos, Dir, self.Owner, self.Bullet, self, self.Direction )
+    if self.ShotCount then
+        for i=0, self.ShotCount do
```

Comment:
```suggestion
        for i=1, self.ShotCount do
```

Reply:
this is incorrect, since 1 == 1 it wont run for a normal weapon with shotcount = 1

Reply:
and plus, it doesnt spawn 2 for a normal weapon

Reply:
Indexes for tables start at 1 in lua, not 0.

Reply:
its not a table...

Reply:
its not incorrect
run `for i=1,1 do print("TEST") end`

Reply:
but i think i know what you mean

Reply:
does a for loop run even if i == themax

Reply:
You're correct, i'm incorrect. lol